nav2_behavior_tree/include/nav2_behavior_tree/clear_costmap_service.hpp

```diff
@@ -39,6 +39,11 @@ class ClearEntireCostmapService : public BtServiceNode<nav2_msgs::srv::ClearEnti
   void on_init() override
   {
   }
+
+  BT::NodeStatus on_send_request_failure() override
+  {
+    return BT::NodeStatus::SUCCESS;
```

Comment:
Returning success on failure is confusing. Why are we doing this, exactly? It seems to me like we need to fix the behavior tree instead.

Reply:
This is being done so that the recovery behaviors an continue even if one of the action servers is not running, i.e. if `Spin` is not available yet, it can attempt to `Wait`. Otherwise, the recovery part of the tree will return a failed status and never execute the remaining behaviors. Also, I think that if the recoveries return a failed status, it won't tick the `NavigateToPose` part of the tree again, @mhpanah  can confirm. I agree that this should eventually be resolved at the behavior tree level, but at the moment I believe that would require a significant change to the current trees and recovery control node, so this is a work-around.

Reply:
Please add a comment/todo mentioning this.

Reply:
I think you want the `ForceSuccessNode` from the BT decorator nodes: https://www.behaviortree.dev/decoratornode/

Reply:
@mkhansen-intel The decorator node would be one solution, though it might be considered a "big hammer". With my change, the recovery action would still be allowed to return failure if it fails to complete its task. My override is really only being used for the case that the send request or send goal itself fails. Adding the decorator node would prevent the recovery actions from ever returning a failure status. In the short term, this might be okay, but realistically either way we'll want to change the bt code as I mentioned before, perhaps part of the updates I listed in #1206.